File: src/main/java/com/tobe/healthy/gym/presentation/GymCommandController.java
package com.tobe.healthy.gym.presentation;

import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.tobe.healthy.ApiResult;
import com.tobe.healthy.config.security.CustomMemberDetails;
import com.tobe.healthy.gym.application.GymCommandService;
import com.tobe.healthy.gym.presentation.dto.in.CommandRegisterGym;
import com.tobe.healthy.gym.presentation.dto.in.CommandSelectMyGym;
import com.tobe.healthy.gym.presentation.dto.out.CommandRegisterGymResult;
import com.tobe.healthy.gym.presentation.dto.out.CommandSelectMyGymResult;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;

@RestController
@RequestMapping("/api/v1/gyms")
@RequiredArgsConstructor
@Tag(name = "04-01.헬스장 API", description = "헬스장 조회 API")
public class GymCommandController {

	private final GymCommandService gymCommandService;

	@Operation(summary = "관리자 또는 트레이너가 헬스장을 등록한다.")
	@PostMapping
	public ApiResult<CommandRegisterGymResult> registerGym(
		@RequestBody CommandRegisterGym request) {
		return ApiResult.success("헬스장을 등록하였습니다.", gymCommandService.registerGym(request));
	}

	@Operation(summary = "학생 또는 트레이너가 내가 다니는 헬스장으로 등록한다.")
	@PostMapping("/{gymId}")
	public ApiResult<CommandSelectMyGymResult> selectMyGym(
		@PathVariable Long gymId,
		@RequestBody(required = false) CommandSelectMyGym request,
		@AuthenticationPrincipal CustomMemberDetails member) {
		return ApiResult.success("내 헬스장으로 등록되었습니다.", gymCommandService.selectMyGym(gymId, request, member.getMemberId()));
	}
}

File: src/main/kotlin/com/tobe/healthy/gym/presentation/GymCommandController.kt
package com.tobe.healthy.gym.presentation

import com.tobe.healthy.ApiResultResponse
import com.tobe.healthy.config.security.CustomMemberDetails
import com.tobe.healthy.gym.application.GymCommandService
import com.tobe.healthy.gym.domain.dto.out.CommandRegisterGymResult
import com.tobe.healthy.gym.domain.dto.out.CommandSelectMyGymResult
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.responses.ApiResponse
import io.swagger.v3.oas.annotations.tags.Tag
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/gyms/v1")
@Tag(name = "04-01.헬스장 API", description = "헬스장 조회 API")
class GymCommandController(
    private val gymCommandService: GymCommandService
) {
    @Operation(
        summary = "관리자 또는 트레이너가 헬스장을 등록한다.",
        description = "관리자 또는 트레이너가 헬스장을 등록한다.",
        responses = [ApiResponse(responseCode = "200", description = "헬스장을 등록하였습니다.")],
    )
    @PostMapping
    fun registerGym(@RequestParam name: String
    ): ApiResultResponse<CommandRegisterGymResult> {
        return ApiResultResponse(
            data = gymCommandService.registerGym(name),
            message = "헬스장을 등록하였습니다."
        )
    }

    @Operation(
        summary = "학생 또는 트레이너가 내가 다니는 헬스장으로 등록한다.",
        description = "학생 또는 트레이너가 내가 다니는 헬스장으로 등록한다.",
        responses = [
            ApiResponse(responseCode = "404", description = "회원을 찾을 수 없습니다."),
            ApiResponse(responseCode = "404", description = "헬스장을 찾을 수 없습니다."),
            ApiResponse(responseCode = "200", description = "내 헬스장으로 등록하였습니다."),
        ],
    )
    @PostMapping("/{gymId}")
    fun selectMyGym(
        @PathVariable gymId: Long,
        joinCode: String?,
        @AuthenticationPrincipal member: CustomMemberDetails,
    ): ApiResultResponse<CommandSelectMyGymResult> {
        return ApiResultResponse(
            data = gymCommandService.selectMyGym(gymId, joinCode, member.memberId),
            message = "내 헬스장으로 등록되었습니다."
        )
    }
}
